### DISCUSSION TITLE
shift_msa, scale_msa, gate_msa, shift_mlp, scale_mlp, gate_mlp = emb.chunk(6, dim=1) ValueError: not enough values to unpack (expected 6, got 4)

### DISCUSSION THREAD

**Discussion by LianShuaiLong** [huggingface/diffusers#9710] [Q&A]
To save GPU memory during training, I pre-extracted the text embeddings. During the training process, I did not load the text encoders and directly used the embeddings for training. However, I encountered the following error:

```
shift_msa, scale_msa, gate_msa, shift_mlp, scale_mlp, gate_mlp = emb.chunk(6, dim=1)
ValueError: not enough values to unpack (expected 6, got 4)
```
Here is my feature extraction script and the modified training script. Could someone kindly help me identify where the issue might be?

**Comment by LianShuaiLong** [comment]
[
[tmp.zip](https://github.com/user-attachments/files/17435044/tmp.zip)
](url)

**Comment by a-r-r-o-w** [comment]
Could you share your code in a public repository instead of a zip file? Which model are you trying to train?

**Reply by LianShuaiLong** [reply]
> Could you share your code in a public repository instead of a zip file? Which model are you trying to train?

flux1-dev,These are the scripts for extracting text embeddings and initiating LoRA training. No other scripts are involved in these two processes.
